pom.xml

```diff
@@ -84,27 +84,37 @@
             <artifactId>data-model-lib</artifactId>
             <groupId>life.qbic</groupId>
             <version>1.11.0</version>
+            <scope>${osgi.scope}</scope>
         </dependency>
         <dependency>
             <artifactId>commons-lang3</artifactId>
             <groupId>org.apache.commons</groupId>
             <version>3.7</version>
+            <scope>${osgi.scope}</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.codehaus.groovy</groupId>
+            <artifactId>groovy</artifactId>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <artifactId>picocli</artifactId>
             <groupId>info.picocli</groupId>
             <version>3.7.0</version>
+            <scope>${osgi.scope}</scope>
         </dependency>
-        <!-- https://mvnrepository.com/artifact/org.apache.httpcomponents/httpclient -->
+        <!-- https://mvnrepository.com/artifact/com.fasterxml.jackson.core/jackson-databind -->
         <dependency>
-            <artifactId>micronaut-http-client</artifactId>
-            <groupId>io.micronaut</groupId>
-            <version>1.2.11</version>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-databind</artifactId>
+            <version>2.12.0</version>
+            <scope>${osgi.scope}</scope>
         </dependency>
         <dependency>
             <artifactId>org.everit.json.schema</artifactId>
             <groupId>com.github.everit-org.json-schema</groupId>
             <version>1.12.1</version>
+            <scope>compile</scope>
```

Comment:
why is this dependency set as compile while other external dependencies are set as osgi.scope?

Reply:
because we always need it it to be available to the classpath during build time, and it is not provided by the OSGi framework